lib/iris/io/format_picker.py

```diff
@@ -214,7 +214,7 @@ def __lt__(self, other):
         if not isinstance(other, FormatSpecification):
             return NotImplemented
 
-        return (-self.priority, hash(self)) < (-other.priority, hash(other))
+        return (-self.priority, self.name) < (-other.priority, other.name)
```

Comment:
Hmm, now that I think about it, this one may only work if names are unique? (because `__eq__` still uses `hash`.)